### DISCUSSION TITLE
#Blog 8

### DISCUSSION THREAD

**Discussion by alex-3709537** [AlinaBoess/SoftwareEngineeringProjektTINF23B5#8] [General]
# Achter Blog - Softwarearchitektur-Dokument

Schön, dass ihr wieder bei uns seid! 😊 Diese Woche haben wir uns intensiv mit dem Softwarearchitektur-Dokument beschäftigt.

Ihr findet es hier: [SAD](https://github.com/AlinaBoess/SoftwareEngineeringProjektTINF23B5/blob/main/documentation/architecture/SAD.md)

Folgende Schritte haben wir behandelt:
### Ziele und Einschränkungen der Architektur

Wir haben die Hauptziele definiert, darunter Sicherheit, Datenschutz, Wiederverwendbarkeit und Verfügbarkeit. Als Einschränkungen gelten die Nutzung spezifischer Technologien, ein kleines Team mit begrenztem Zeitrahmen und die Integration von Legacy Code.
### Prozessansicht

Anhand von Sequenzdiagrammen zeigen wir, wie Benutzer, Systemkomponenten und Datenbanken bei zentralen Anwendungsfällen wie Login, Reservierung, Feedback und Verwaltung interagieren.
### Implementierungsansicht

Diese beschreibt die Systemstruktur, die in Schichten oder Subsysteme organisiert ist, und erläutert die Zuordnung der Funktionen zu diesen Bereichen.
### Qualität

Wir haben wichtige Qualitätsaspekte betrachtet, wie Erweiterbarkeit, Verlässlichkeit, Übertragbarkeit und Sicherheit. Der modulare Aufbau, der Einsatz moderner Technologien und ein iterativer Entwicklungsprozess sorgen dafür, dass das System langfristig wartbar und anpassbar bleibt.

Wir hoffen, die oben beschriebenen Entscheidungen sind nachvollziehbar! Schaut gerne nächste Woche wieder vorbei, und lasst uns wissen, was ihr denkt oder falls ihr noch Fragen zu unserem SAD habt. 😊

Liebe Grüße
Euer Team: Lukas, Alex, Moumen, Yahya und Alina

**Comment by lucariebel** [comment]
Hi,

euer SAD ist sehr gut strukturiert und die Sequenzdiagramme sind anschaulich und hilfreich. Besonders gut finde ich die klare Darstellung der Architekturziele und den Fokus auf Wartbarkeit.

Was wollt ihr verwenden um auf die 99,5% Systemverfügbarkeit kommen?

Viele Grüße
euer WebShop24

**Comment by SafaeKa** [comment]
Hallo,

euer SAD sieht wirklich klasse aus, und die Sequenzdiagramme sind sehr detailliert. Man kann alle nötigen Informationen für das Verständnis des Projekts daraus entnehmen.

Ich hätte nur eine Frage zum Login: Habt ihr das "Registrieren"-Feature seit dem letzten Mal doch implementiert? Falls nicht, plant ihr, das später noch zu machen?

Viele Grüße,
GaLiGrü 😊
Paula, Safae, Samuel, Valentin, Jonas

>Green Sprout 🌱

**Comment by GermanJesus-lul** [comment]
Hey,

Das SAD sieht gut aus. Mir gefällt vor allem, dass ihr auch über die Anwendung direkt hinausgeht und Entwicklungstools, wie Jira und GitHub, in die Ausarbeitung einbezieht.
 
 Mit freundlichen Grüßen,
Julius und das Rezeptbuch Team
